src/cephadm/cephadm

```diff
@@ -2370,6 +2401,12 @@ def command_adopt():
         call_throws([container_path, 'pull', args.image])
 
     (daemon_type, daemon_id) = args.name.split('.', 1)
+
+    if daemon_type == 'prometheus':
+        (uid, gid) = extract_uid_gid_monitoring(daemon_type)
+        command_adopt_prometheus(daemon_id, uid, gid)
+        return
+
```

Comment:
this ignores the `--style legacy` arg ... 

move this logic down into the `if` statement below or add handling for it in the `command_adopt_prometheus` func?

Reply:
it is in command_adopt_prometheus... line 2543.